sphinx/writers/latex.py

```diff
@@ -1444,7 +1445,9 @@ def visit_entry(self, node):
             self.needs_linetrimming = 1
         if len(node.traverse(nodes.paragraph)) >= 2:
             self.table.has_oldproblematic = True
-        if isinstance(node.parent.parent, nodes.thead):
+        if isinstance(node.parent.parent, nodes.thead) or \
+           node.parent.parent.parent.stubs[cell.col]:
+            # "stubs" list is an attribute of the tgroup element
             if len(node) == 1 and isinstance(node[0], nodes.paragraph) and node.astext() == '':
                 pass
             else:
```

Comment:
It's tough problem.
Off the top of my head, there are two ways: 
1) bringing breaking change to `\sphinxstylethead`. It was introduced at Sphinx-1.5, so nobody (or only very few people) customizes it.
2) add a new macro: `\sphinxstyletheadfamily`